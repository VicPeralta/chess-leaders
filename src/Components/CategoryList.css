.category-container {
  display: flex;
  flex-wrap: wrap;
  gap: 8px;
  width: 100%;
  justify-content: center;
}

.search-bar {
  display: flex;
  align-items: center;
  width: 90%;
  margin: 10px auto;
}

.search {
  width: 90%;
  margin-left: 3px;
  border: none;
  padding: 3px 8px;
}

.category-container a {
  color: var(--text-color);
  text-decoration: none;
}

a:nth-child(4n + 2) {
  background-color: var(--alternate-color);
}

a:nth-child(4n + 3) {
  background-color: var(--alternate-color);
}
