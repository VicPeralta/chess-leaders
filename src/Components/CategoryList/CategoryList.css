.category-container {
  display: flex;
  flex-wrap: wrap;
  gap: 8px;
  width: 100%;
  justify-content: center;
}

.category-container a {
  color: var(--text-color);
  text-decoration: none;
}

a:nth-child(4n + 2) {
  background-color: var(--alternate-color);
}

a:nth-child(4n + 3) {
  background-color: var(--alternate-color);
}

@media (min-width: 513px) {
  a:nth-child(even) {
    background-color: var(--alternate-color);
  }

  a:nth-child(odd) {
    background-color: var(--bg-color);
  }
}
