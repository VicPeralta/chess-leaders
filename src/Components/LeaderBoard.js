import { useParams, Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { IoMdArrowRoundBack } from 'react-icons/io';

const LeaderBoard = () => {
  const params = useParams();
  const { index } = params;
  const leaders = useSelector((state) => (state.chessData[index].data));
  return (
    <>
      <Link to="/">
        <IoMdArrowRoundBack />
      </Link>
      <div>
        {leaders.map((leader) => (
          <Link key={leader.username} to={`/playercard/${leader.username}/${index}`}>
            <p>
              {leader.username}
            </p>
          </Link>
        ))}
      </div>
    </>
  );
};

export default LeaderBoard;
