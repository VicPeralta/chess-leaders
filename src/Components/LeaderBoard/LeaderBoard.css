.leader-container {
  width: 90%;
  margin-inline: auto;
}

.leader-container > .leader-row:nth-child(even) {
  background-color: var(--alternate-color);
}
