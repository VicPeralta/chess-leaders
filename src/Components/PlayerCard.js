import { useParams } from 'react-router';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getPlayerData } from '../Redux/chessLeaders/chess';
import Fetching from './Fetching';
import SearchBar from './SearchBar';
import noPhoto from '../assets/noPhoto.svg';
import './PlayerCard.css';

const PlayerCard = () => {
  const params = useParams();
  const { index } = params;
  const { username } = params;
  const dispath = useDispatch();
  useEffect(() => (dispath(getPlayerData(username))), []);
  const fetching = useSelector((state) => (state.fetching));
  const chessData = useSelector((state) => (state.chessData[index].data));
  const playerData = useSelector((state) => (state.playerData));
  const generalInfo = chessData.find((player) => (player.username === username));
  const dummySearch = () => {
  };
  const back = `/leaderboard/${index}`;
  return (
    <>
      <SearchBar
        searchHandler={dummySearch}
        backText="BACK"
        backPath={back}
        searchText=" "
      />

      {
        fetching && (
          <>
            <Fetching />
          </>
        )
      }
      {
        !fetching && playerData && (
          <div className="player-info">
            {playerData.avatar && (<img src={playerData.avatar} alt="Player avatar" />)}
            {!playerData.avatar && (<img src={noPhoto} alt="Player avatar" />)}
            <p className="player-name">
              {playerData.name}
            </p>
            <table>
              <tr>
                <td>Rank #:</td>
                <td>{generalInfo.rank}</td>
              </tr>
              <tr>
                <td>User name:</td>
                <td>{playerData.username}</td>
              </tr>
              <tr>
                <td>Title:</td>
                <td>{playerData.title}</td>
              </tr>
              <tr>
                <td>Score:</td>
                <td>{generalInfo.score}</td>
              </tr>
              <tr>
                <td>Games won:</td>
                <td>{generalInfo.win_count}</td>
              </tr>
              <tr>
                <td>Games lost:</td>
                <td>{generalInfo.loss_count}</td>
              </tr>
              <tr>
                <td>Games drawn:</td>
                <td>{generalInfo.draw_count}</td>
              </tr>
              <tr>
                <td>Location:</td>
                <td>{playerData.location}</td>
              </tr>
            </table>
          </div>
        )
      }
    </>
  );
};

export default PlayerCard;
