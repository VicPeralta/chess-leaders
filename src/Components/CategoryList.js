import { useSelector } from 'react-redux';
import { useState } from 'react';

import Fetching from './Fetching';
import Category from './CategoryCard';
import './CategoryList.css';
import SearchBar from './SearchBar';

const CategoryList = () => {
  const fetching = useSelector((state) => (state.fetching));
  const chessData = useSelector((state) => (state.chessData));
  const [filter, setFilter] = useState();
  const handleFilter = (value) => {
    setFilter(value);
  };
  return (
    <>
      {fetching && (<Fetching />)}
      {!fetching && chessData && (
        <>
          <SearchBar searchHandler={handleFilter} backText="" backPath="/" searchText="Categories" />
          <div className="category-container">
            {!filter && (
              <>
                {
                  chessData.map((category) => (
                    <Category
                      key={category.id}
                      id={category.id}
                      icon={category.name}
                      name={category.name}
                      average={category.average}
                    />
                  ))
                }
              </>
            )}
            {filter && (
              <>
                {
                  chessData.filter((category) => (category.name.includes(filter)))
                    .map((category) => (
                      <Category
                        key={category.id}
                        id={category.id}
                        icon={category.name}
                        name={category.name}
                        average={category.average}
                      />
                    ))
                }
              </>
            )}
          </div>
        </>
      )}
    </>
  );
};

export default CategoryList;
