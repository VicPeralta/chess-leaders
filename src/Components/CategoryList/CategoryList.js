import { useSelector } from 'react-redux';
import { useState } from 'react';

import Fetching from '../Fetching/Fetching';
import Category from '../CategoryCard/CategoryCard';
import SearchBar from '../SearchBar/SearchBar';
import './CategoryList.css';

const CategoryList = () => {
  const fetching = useSelector((state) => (state.fetching));
  const chessData = useSelector((state) => (state.chessData));
  const [filter, setFilter] = useState();
  const handleFilter = (value) => {
    setFilter(value);
  };
  return (
    <>
      {fetching && (<Fetching />)}
      {!fetching && chessData && (
        <>
          <SearchBar searchHandler={handleFilter} backText="" backPath="/" searchText="Categories" />
          <div className="category-container">
            {!filter && (
              <>
                {
                  chessData.map((category) => (
                    <Category
                      key={category.id}
                      id={category.id}
                      icon={category.name}
                      name={category.name}
                      average={category.average}
                    />
                  ))
                }
              </>
            )}
            {filter && (
              <>
                {
                  chessData.filter((category) => (category.name.includes(filter)))
                    .map((category) => (
                      <Category
                        key={category.id}
                        id={category.id}
                        icon={category.name}
                        name={category.name}
                        average={category.average}
                      />
                    ))
                }
              </>
            )}
          </div>
        </>
      )}
    </>
  );
};

export default CategoryList;
