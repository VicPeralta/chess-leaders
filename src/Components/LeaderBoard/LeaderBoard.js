import { useParams, Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useState } from 'react';
import SearchBar from '../SearchBar/SearchBar';
import Category from '../CategoryCard/CategoryCard';
import Leader from '../Leader/Leader';
import './LeaderBoard.css';

const LeaderBoard = () => {
  const params = useParams();
  const { index } = params;
  const [filter, setFilter] = useState();
  const leaders = useSelector((state) => (state.chessData[index]));
  const filterHandler = (value) => {
    setFilter(value);
  };

  return (
    <>
      <SearchBar backText="HOME" backPath="/" searchHandler={filterHandler} searchText="Player" />
      <Category
        id={Number(index)}
        name={leaders.name}
        average={leaders.average}
        detailButton={false}
      />
      {!filter && (
        <div className="leader-container">
          {
            leaders.data.map((leader) => (
              <Leader
                key={leader.rank}
                index={Number(index)}
                rank={leader.rank}
                username={leader.username}
              />
            ))
          }
        </div>
      )}
      {filter && (
        <div className="leader-container">
          {
            leaders.data.filter((leader) => (leader.username.includes(filter)))
              .map((leader) => (
                <Link key={leader.username} to={`/playercard/${leader.username}/${index}`}>
                  <p>
                    {leader.username}
                  </p>
                </Link>
              ))
          }
        </div>
      )}
    </>
  );
};

export default LeaderBoard;
