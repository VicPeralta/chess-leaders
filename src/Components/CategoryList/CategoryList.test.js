import { Provider } from 'react-redux';
import { Route, Routes } from 'react-router';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import store from '../../Redux/configureStore';
import CategoryList from './CategoryList';
import LeaderBoard from '../LeaderBoard/LeaderBoard';
import PlayerCard from '../PlayerCard/PlayerCard';

jest.mock('../../Redux/chessLeaders/chess.js');

describe('CategoryList Component Test', () => {
  const element = (
    <MemoryRouter>
      <Provider store={store}>
        <Routes>
          <Route path="/" element={<CategoryList />} />
          <Route path="/leaderboard" element={<LeaderBoard />}>
            <Route path=":index" element={<LeaderBoard />} />
          </Route>
          <Route path="/playerCard" element={<PlayerCard />}>
            <Route path=":username/:index" element={<PlayerCard />} />
          </Route>
        </Routes>
      </Provider>
    </MemoryRouter>
  );
  test('Match Snapshot', () => {
    const list = renderer.create(element);
    expect(list).toMatchSnapshot();
  });
  test('Render correctly', () => {
    render(element);
    const list = screen.getByText(/Special category/);
    expect(list).toBeInTheDocument();
  });
  test('User interaction', () => {
    render(element);
    // The second link is the first card
    const card = screen.getAllByRole('link')[1];
    expect(card).toBeInTheDocument();
    // From the category list, navigates to leaderboard
    userEvent.click(card);
    // The list of players includes VicPeralta & kasparov
    expect(screen.getByText(/VicPeralta/)).toBeInTheDocument();
    expect(screen.getByText(/kasparov/)).toBeInTheDocument();
  });
});
