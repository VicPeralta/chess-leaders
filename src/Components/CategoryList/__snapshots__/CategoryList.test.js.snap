// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList Component Test Rendering CategoryList Fetching 1`] = `
Array [
  <div
    className="search-bar"
    data-testid="searchBar-element"
  >
    <div
      className="back-section"
    >
      <a
        href="#/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </a>
      <span
        className="back-text"
      />
    </div>
    <input
      className="search"
      onChange={[Function]}
      placeholder="Categories"
      type="text"
    />
    <div
      className="icons"
    >
      <svg
        fill="currentColor"
        height={20}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "margin": "0 10px",
          }
        }
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 12V6c0-2.217-1.785-4.021-3.979-4.021a.933.933 0 0 0-.209.025A4.006 4.006 0 0 0 8 6v6c0 2.206 1.794 4 4 4s4-1.794 4-4zm-6 0V6c0-1.103.897-2 2-2a.89.89 0 0 0 .163-.015C13.188 4.06 14 4.935 14 6v6c0 1.103-.897 2-2 2s-2-.897-2-2z"
        />
        <path
          d="M6 12H4c0 4.072 3.061 7.436 7 7.931V22h2v-2.069c3.939-.495 7-3.858 7-7.931h-2c0 3.309-2.691 6-6 6s-6-2.691-6-6z"
        />
      </svg>
      <svg
        fill="currentColor"
        height={20}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a443.74 443.74 0 0 0-79.7-137.9l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.4a351.86 351.86 0 0 0-99 57.4l-81.9-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a446.02 446.02 0 0 0-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0 0 25.8 25.7l2.7.5a449.4 449.4 0 0 0 159 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-85a350 350 0 0 0 99.7-57.6l81.3 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 0 1-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 0 1-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 0 1 512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 0 1 400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 0 1 624 502c0 29.9-11.7 58-32.8 79.2z"
        />
      </svg>
    </div>
  </div>,
  <div
    className="category-container"
    data-testid="categoryList-component"
  >
    <a
      href="#/leaderboard/0"
      onClick={[Function]}
    >
      <div
        className="category-card"
        data-id={0}
        role="presentation"
      >
        <div
          className="category-title-section"
        >
          <svg
            fill="currentColor"
            height={24}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
            />
            <path
              d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
            />
          </svg>
        </div>
        <img
          alt="the category"
          src="daily.png"
        />
        <p
          className="category-title"
        >
          daily
        </p>
        <div
          className="average"
        >
          <span
            className="highlighted"
          >
            avg:  
          </span>
          <span>
            2538.5
          </span>
        </div>
      </div>
    </a>
    <a
      href="#/leaderboard/1"
      onClick={[Function]}
    >
      <div
        className="category-card"
        data-id={1}
        role="presentation"
      >
        <div
          className="category-title-section"
        >
          <svg
            fill="currentColor"
            height={24}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
            />
            <path
              d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
            />
          </svg>
        </div>
        <img
          alt="the category"
          src="daily960.png"
        />
        <p
          className="category-title"
        >
          daily960
        </p>
        <div
          className="average"
        >
          <span
            className="highlighted"
          >
            avg:  
          </span>
          <span>
            2538.5
          </span>
        </div>
      </div>
    </a>
  </div>,
]
`;

exports[`CategoryList Component Test Rendering CategoryList Success 1`] = `
Array [
  <div
    className="search-bar"
    data-testid="searchBar-element"
  >
    <div
      className="back-section"
    >
      <a
        href="#/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </a>
      <span
        className="back-text"
      />
    </div>
    <input
      className="search"
      onChange={[Function]}
      placeholder="Categories"
      type="text"
    />
    <div
      className="icons"
    >
      <svg
        fill="currentColor"
        height={20}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "margin": "0 10px",
          }
        }
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 12V6c0-2.217-1.785-4.021-3.979-4.021a.933.933 0 0 0-.209.025A4.006 4.006 0 0 0 8 6v6c0 2.206 1.794 4 4 4s4-1.794 4-4zm-6 0V6c0-1.103.897-2 2-2a.89.89 0 0 0 .163-.015C13.188 4.06 14 4.935 14 6v6c0 1.103-.897 2-2 2s-2-.897-2-2z"
        />
        <path
          d="M6 12H4c0 4.072 3.061 7.436 7 7.931V22h2v-2.069c3.939-.495 7-3.858 7-7.931h-2c0 3.309-2.691 6-6 6s-6-2.691-6-6z"
        />
      </svg>
      <svg
        fill="currentColor"
        height={20}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a443.74 443.74 0 0 0-79.7-137.9l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.4a351.86 351.86 0 0 0-99 57.4l-81.9-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a446.02 446.02 0 0 0-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0 0 25.8 25.7l2.7.5a449.4 449.4 0 0 0 159 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-85a350 350 0 0 0 99.7-57.6l81.3 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 0 1-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 0 1-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 0 1 512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 0 1 400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 0 1 624 502c0 29.9-11.7 58-32.8 79.2z"
        />
      </svg>
    </div>
  </div>,
  <div
    className="category-container"
    data-testid="categoryList-component"
  >
    <a
      href="#/leaderboard/0"
      onClick={[Function]}
    >
      <div
        className="category-card"
        data-id={0}
        role="presentation"
      >
        <div
          className="category-title-section"
        >
          <svg
            fill="currentColor"
            height={24}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
            />
            <path
              d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
            />
          </svg>
        </div>
        <img
          alt="the category"
          src="daily.png"
        />
        <p
          className="category-title"
        >
          daily
        </p>
        <div
          className="average"
        >
          <span
            className="highlighted"
          >
            avg:  
          </span>
          <span>
            2538.5
          </span>
        </div>
      </div>
    </a>
    <a
      href="#/leaderboard/1"
      onClick={[Function]}
    >
      <div
        className="category-card"
        data-id={1}
        role="presentation"
      >
        <div
          className="category-title-section"
        >
          <svg
            fill="currentColor"
            height={24}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
            />
            <path
              d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
            />
          </svg>
        </div>
        <img
          alt="the category"
          src="daily960.png"
        />
        <p
          className="category-title"
        >
          daily960
        </p>
        <div
          className="average"
        >
          <span
            className="highlighted"
          >
            avg:  
          </span>
          <span>
            2538.5
          </span>
        </div>
      </div>
    </a>
  </div>,
]
`;
